test/function/param.cpp

```diff
@@ -235,19 +236,16 @@ namespace {
   TEST( LoadParamTest, InvalidName ) {
     struct stumpless_param param;
     const struct stumpless_param *result;
+    stumpless_test_data invalid_names = load_corpus_folder("invalid_param_name");
 
-    result = stumpless_load_param( &param, "par=am", "test-value" );
-    EXPECT_NULL( result );
-    EXPECT_ERROR_ID_EQ( STUMPLESS_INVALID_ENCODING );
-
-    result = stumpless_load_param( &param, "param]", "test-value" );
-    EXPECT_NULL( result );
-    EXPECT_ERROR_ID_EQ( STUMPLESS_INVALID_ENCODING );
+    for(int i = 0; i < invalid_names.length; ++i) {
+      result = stumpless_load_param( &param, invalid_names.test_strings[i], "test-value" );
+      EXPECT_NULL( result );
+      EXPECT_ERROR_ID_EQ( STUMPLESS_INVALID_ENCODING );
+      delete[] invalid_names.test_strings[i];
```

Comment:
As you refactor to use a C++ vector (or other structure), do your best to make it easier for these tests to run by not requiring manual deletion of strings.